agreement/events.go

```diff
@@ -407,6 +411,36 @@ func (e newRoundEvent) ComparableStr() string {
 	return e.String()
 }
 
+type readLowestEvent struct {
+	// T currently only supports readLowestVote
+	T eventType
+
+	// Round and Period are the round and period for which to query the
+	// lowest-credential vote, value or payload.  This type of event is only
+	// sent for reading the lowest period 0 credential, but the Period is here
+	// anyway to route to the appropriate proposalMachinePeriod.
+	Round  round
+	Period period
```

Comment:
How does this Period=0 thing working? I don't see how the calling code can work generically across event types, so it can't just pluck `ev.Period` out of an arbitrary event of any type.